tests/test/misc.rs

```diff
@@ -727,3 +727,34 @@ fn canonicalization_regression() {
         }
     }
 }
+
+#[test]
+#[ignore]
+// this is a regression test for an infinite loop, but it doesn't actually work
+// because the code fails coherence checking.
```

Comment:
hmm I feel like we *used* to have a way (and should have a way) to disable coherence, I guess that would help?

Reply:
It would, yeah. (I tried the test manually by disabling the coherence check in `checked_program`.) I didn't find such a way, but I could try adding it.

Reply:
I think adding it makes sense, not sure how annoying it would be to do though. Maybe it has to be an "input" into the salsa setup?